internal/clients/client_models.go

```diff
@@ -30,4 +30,9 @@ type InstanceType struct {
 
 	// The size of the memory, in MiB.
 	Memory int64 `json:"memory,omitempty"`
+
+	// Does the instance type supports RHEL
+	Supported bool `json:"supported,omitempty"`
+
+	Architecture string `json:"architecture,omitempty"`
 }
```

Comment:
This could be a type alias: `type Architecture string` with constant strings for both two architectures: x86_64 and ARM. But I think this could create a problem when marhsalling to JSON and you would need to implement `MarshalJSON() ([]byte, error)` function that converts it to string (which is pretty easy). Added benefit would be that no one could ever assign anything else than the two constants here. But feel free to keep it as a string.

As I commented above, I think one last field which is worth adding is `InstanceStorageMiB integer` (name and type would depend on what AWS library uses). It should be directly readable and for EBS-only types we would set this to zero. Not insisting tho.

Please always add code comments for important types like this one. Here, I would definitely drop a comment about what is the expected format of architecture.

Reply:
I will add the storage in a different PR